tests/scripts/all.sh

```diff
@@ -6077,6 +6077,14 @@ support_build_cmake_custom_config_file () {
     support_test_cmake_out_of_source
 }
 
+component_build_cmake_programs_no_testing () {
+    msg "build: cmake with -DENABLE_PROGRAMS=ON and -DENABLE_TESTING=OFF"
```

Comment:
It might be useful to have a comment as to why we want to check this

```
# Verifying that the type of builds performed by oss-fuzz don't get accidentally broken
```